### PR TITLE
Fixes to iOS plugin

### PR DESCRIPTION
**Pull request by emilsas**
**Updates:**
- Fix issue #23 
- Update google-cast-sdk to v4.4.4 due to Google notice https://developers.google.com/cast/docs/ios_sender/ios13_changes
- Typo in chrome.cast.js in textTrackStyle

@ghenry22 now is ready for testing ;)

### CODE REVIEW

**Comment by ghenry22** [comment]
@emilsas oh nice one! I will fire up my test app tomorrow and provide feedback on this!

**Comment by emilsas** [comment]
Hi @ghenry22,
Any updates?

**Comment by emilsas** [comment]
**Updates: (EDITED)**
- Fix issue #23
- Update google-cast-sdk to v4.4.4 due to Google notice https://developers.google.com/cast/docs/ios_sender/ios13_changes
- Typo in chrome.cast.js in textTrackStyle
- Call calback error when the user tap cancel in "Cast to" dialog. Same as on android.

**Comment by dkanada** [comment]
@ghenry22 any updates here?

File: src/ios/CastUtilities.swift [hunk 1]
```diff
@@ -80,17 +82,21 @@ class CastUtilities {
                 mediaTextTrackStyle.windowType = parseWindowType(windowType)
             }
 
+            return mediaTextTrackStyle
+        } else {
+            return nil
         }
 
-        return mediaTextTrackStyle
+
```
Comment (dkanada) [inline src/ios/CastUtilities.swift:90]: remove two extra newlines

File: src/ios/CastUtilities.swift [hunk 2]
```diff
@@ -26,12 +26,14 @@ class CastUtilities {
...
 
-    static func buildTextTrackStyle(_ data: Data) -> GCKMediaTextTrackStyle {
+    static func buildTextTrackStyle(_ data: Data) -> GCKMediaTextTrackStyle? {
         let json = try? JSONSerialization.jsonObject(with: data, options: [])
 
-        let mediaTextTrackStyle = GCKMediaTextTrackStyle.createDefault()
+        let dict = json as? [String: Any] ?? [:]
+
+        if (dict.count != 0) {
+            let mediaTextTrackStyle = GCKMediaTextTrackStyle.createDefault()
```
Comment (dkanada) [inline src/ios/CastUtilities.swift:35]: This looks like an improvement, I think we can merge regardless.

File: chrome.cast.js [hunk 3]
```diff
@@ -1126,6 +1126,7 @@ chrome.cast.media.Media.prototype._update = function(isAlive, obj) {
 	this.mediaSessionId = obj.mediaSessionId || this.mediaSessionId;
 	this.playbackRate = obj.playbackRate || this.playbackRate;
 	this.playerState = obj.playerState || this.playerState;
+  this.activeTrackIds = obj.activeTrackIds || this.activeTrackIds;
```
Comment (dkanada) [inline chrome.cast.js:1129]: indentation

File: plugin.xml [hunk 4]
```diff
@@ -62,7 +62,7 @@
         <source url="https://github.com/CocoaPods/Specs.git"/>
       </config>
       <pods use-frameworks="true">
-        <pod name="google-cast-sdk" spec="~> 4.4.1" />
+        <pod name="google-cast-sdk" spec="~> 4.4.4" />
```
Comment (dkanada) [inline plugin.xml:65]: This won't break the Java implementation will it?

Comment (Lindsay-Needs-Sleep) [reply plugin.xml:65]: This has no affect on java/android at all

**Pull request closed without merging**